### PR TITLE
docs(readme): add types

### PR DESCRIPTION
**Pull request by harish-sethuraman**
This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case

Added types in Readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,26 +112,43 @@ Works like [`output.filename`](https://webpack.js.org/configuration/output/#outp
...
 Remove Order Warnings.
 See [examples](#remove-order-warnings) below for details.
 
 #### `insert`
 
-Type: `String|Function`
+Type:
+
+```ts
+type insert = string | ((linkTag: any) => void);
```
Comment (alexander-akait) [inline README.md:149]: Please avoid using `any`, if you found it in our code, it means bug, we should improve this type

File: README.md [hunk 2]
```diff
@@ -173,7 +201,12 @@ A new `<link>` element will be inserted before the element with id `some-element
 
 #### `attributes`
 
-Type: `Object`
+Type:
+
+```ts
+type attributes = object;
```
Comment (alexander-akait) [inline README.md:207]: I think it is `Record<string, string>`

File: README.md (hunk 1 above)
Comment (harish-sethuraman) [reply README.md:149]: The linkTag here is an `HTMLElement` right?

Comment (alexander-akait) [reply README.md:149]: I don't know, but mostly yes, just check it

Comment (harish-sethuraman) [reply README.md]: Changed to `HTMLLinkElement`

File: README.md (hunk 2 above)
Comment (harish-sethuraman) [reply README.md]: Changed

File: README.md [hunk 3]
```diff
@@ -31,10 +31,16 @@ Compared to the extract-text-webpack-plugin:
...
 
-```bash
+```console
 npm install --save-dev mini-css-extract-plugin
 ```
 
+or
+
+```console
+yarn add -D mini-css-extract-plugin
```
Comment (alexander-akait) [inline README.md:41]: Change it on `npm`, because we use npm

Comment (alexander-akait) [reply README.md:41]: If we want to provide yarn, we should also provide pnpm example, i.e. for all package managers

Comment (harish-sethuraman) [reply README.md:41]: 🤔  I've seen npm and yarn in some repos. That is why I started adding yarn in all repos. Should we consider adding `pnpm` as well?

Comment (alexander-akait) [reply README.md:41]: If we add `yarn` we should add `pnpm` too

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
Thanks

**Pull request merged**